### PR TITLE
Support Grouping translation in YANG Models.

### PR DESCRIPTION
**Pull request by praveen-li**
[sonic_yang_ext.py]: Support Grouping translation in YANG Models.
Changes:
 -- pre Process Grouping section from all yang models, so it can be used
from any yang model.
-- add jsondiff in setup.py, it is useful for test debugging in case of
failures.
-- use 'stypes' instead of head.
-- pass config DB table name in _createLeafDict().
-- added test config for grouping.
-- white spaces changes.

Note: Changes are done in the way that we can add supoort for other
Generic YANG statement easily for tranlation.

Signed-off-by: Praveen Chaudhary pchaudhary@linkedin.com

#### Why I did it

Need to support Grouping translation in YANG Models.

#### How I did it

Changes:
 -- pre Process Grouping section from all yang models, so it can be used
from any yang model. File: src/sonic-yang-mgmt/sonic_yang_ext.py

-- add jsondiff in setup.py, it is useful for test debugging in case of
failures.
-- use 'stypes' instead of head.
-- pass config DB table name in _createLeafDict(). 
-- added test config for grouping.
-- white spaces changes.

#### How to verify it

Added Build Time Tests: File: src/sonic-yang-models/tests/files/sample_config_db.json

Build the PKG.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Comment by praveen-li** [comment]
@jleveque : can you help in running the Azure pipeline again, seems like temporary failure. Thanks a lot.

**Comment by praveen-li** [comment]
@qiluo-msft, @renukamanavalan : not sure, who is the right person to review this PR.
But since it unblocks couple of yang model PRs, and is a useful feature in Yang. So Kindly  assign someone to review it. Thanks a ton for help.

**Comment by qiluo-msft** [comment]
@arlakshm Could you help review?

**Comment by praveen-li** [comment]
Rebased to rerun Azure Pipeline

**Comment by qiluo-msft** [comment]
PR owner could use `/apzw run` to trigger the Azure pipeline build checkers.
This feature is enabled on this repo and several other sonic repoes.

File: src/sonic-yang-mgmt/sonic_yang_ext.py [hunk 1]
```diff
@@ -245,7 +390,7 @@ def _yangConvert(val):
             elif 'leafref' in type:
                 vValue = val
             #TODO: find type in sonic-head, as of now, all are enumeration
-            elif 'head:' in type:
+            elif 'stypes:' in type:
```
Comment (arlakshm) [inline src/sonic-yang-mgmt/sonic_yang_ext.py:393]: Is this to get the enum define in `sonic-types` yang model?

Comment (praveen-li) [reply src/sonic-yang-mgmt/sonic_yang_ext.py:393]: yes this is for definitions in stypes

**Review by arlakshm — APPROVED** [review]
lgtm

**Pull request merged**